django/utils/text.py

```diff
@@ -314,8 +316,27 @@ def phone2numeric(phone):
     return "".join(char2number.get(c, c) for c in phone.lower())
 
 
-def compress_string(s):
-    return gzip_compress(s, compresslevel=6, mtime=0)
+def _get_random_fname(max_random_bytes):
+    return "a" * secrets.randbelow(max_random_bytes)
+
+
+def compress_string(s, max_random_bytes=0):
+    compressed_data = gzip_compress(s, compresslevel=6, mtime=0)
+
+    if max_random_bytes == 0:
+        return compressed_data
+
+    header = bytearray(compressed_data[:10])
+    header[3] = gzip.FNAME
+
+    return b"".join(
+        [
+            header,
+            _get_random_fname(max_random_bytes).encode("ascii"),
```

Comment:
```suggestion
            _get_random_fname(max_random_bytes),
```

Reply:
This would also be less clunky styled as:
```python
    fname = _get_random_fname(max_random_bytes) + b"\x00"
    return b"".join([header, fname, compressed_data[10:]])
```